<a href="https://colab.research.google.com/github/piutrus/kurs-ai/blob/main/pandas2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd

df = pd.read_csv('Pokemon.csv')
df

In [6]:
df['Type 1'].nunique()

18

In [9]:
df['Legendary'].sum()
print(df['Legendary'].value_counts()[True])
print(len(df[df['Legendary'] == True]))

65
65


In [16]:
print(df['Type 2'].isna().sum())

386


In [19]:
df.sort_values(by = 'Total', ascending = False)

#                   Name   Type 1    Type 2  Total   HP  Attack  \
426  384  RayquazaMega Rayquaza   Dragon    Flying    780  105     180   
163  150    MewtwoMega Mewtwo X  Psychic  Fighting    780  106     190   
164  150    MewtwoMega Mewtwo Y  Psychic       NaN    780  106     150   
422  382    KyogrePrimal Kyogre    Water       NaN    770  100     150   
424  383  GroudonPrimal Groudon   Ground      Fire    770  100     180   
..   ...                    ...      ...       ...    ...  ...     ...   
13    10               Caterpie      Bug       NaN    195   45      30   
16    13                 Weedle      Bug    Poison    195   40      35   
446  401              Kricketot      Bug       NaN    194   37      25   
322  298                Azurill   Normal     Fairy    190   50      20   
206  191                Sunkern    Grass       NaN    180   30      30   

     Defense  Sp. Atk  Sp. Def  Speed  Generation  Legendary  
426      100      180      100    115           3       True  
163      100      154      100    130           1       True  
164       70      194      120    140           1       True  
422       90      180      160     90           3       True  
424      160      150       90     90           3       True  
..       ...      ...      ...    ...         ...        ...  
13        35       20       20     45           1      False  
16        30       20       20     50           1      False  
446       41       25       41     25           4      False  
322       40       20       40     20           3      False  
206       30       30       30     30           2      False  

[800 rows x 13 columns]

1. Wczytaj plik “Pokemon.csv” i wyświetl pierwsze 10 rekordów
2. Wyświetl liczbę unikalnych wartości w kolumnie “Type 1” *
3. Policz ile jest pokemonów legendarnych
4. Sprawdź ile pokemonów nie posiada Typu 2
5. Znajdź pokemony z najwyższą wartością Total
6. Zlicz pokemony z każdej generacji (Generation) i narysuj wykres kołowy
prezentujący ich liczbę
7. Znajdź wszystkie pokemony, których główny typ (Type 1) to ogień
8. Wyświetl Pokémony, które mają HP większe niż 100
9. Wyświetl legendarne pokemony z Speed powyżej 120
10. Wyświetl pokemony które mają zarówno Attack > 100, jak i Defense < 50
11. Znajdź średnią wartość HP dla każdego z typów pokemonów

12.Oblicz średnią moc (Total) dla pokemonów legendarnych i
nie-legendarnych
13.Oblicz średnią prędkość (Speed) dla każdej z generacji